File: app/src/main/java/com/example/restaurantapp/ui/Customer/CustomerLoginActivity.java
package com.example.restaurantapp.ui.Customer;

import android.content.Intent;
import android.os.Bundle;
import android.text.TextUtils;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import com.example.restaurantapp.R;
import com.example.restaurantapp.data.api.UserApi;
import com.example.restaurantapp.model.User;

import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;

public class CustomerLoginActivity extends AppCompatActivity {

    private final ExecutorService executor = Executors.newSingleThreadExecutor();

    private ImageView btnBack;
    private EditText edtEmail;
    private EditText edtPassword;
    private Button btnContinue;
    private TextView txtCreateAccount;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_customer_login);

        btnBack = findViewById(R.id.btnBack);
        edtEmail = findViewById(R.id.edtCustomerEmail);
        edtPassword = findViewById(R.id.edtCustomerPassword);
        btnContinue = findViewById(R.id.btnCustomerContinue);
        txtCreateAccount = findViewById(R.id.txtCreateAccount);

        btnBack.setOnClickListener(v -> finish());

        String prefill = getIntent().getStringExtra("PREFILL_USERNAME");
        if (prefill != null) {
            edtEmail.setText(prefill);
        }

        txtCreateAccount.setOnClickListener(v -> {
            startActivity(new Intent(this, CustomerCreateAccountActivity.class));
        });

        btnContinue.setOnClickListener(v -> {
            String email = edtEmail.getText().toString().trim();
            String password = edtPassword.getText().toString();

            if (TextUtils.isEmpty(email) || TextUtils.isEmpty(password)) {
                Toast.makeText(this, "Please enter email and password", Toast.LENGTH_SHORT).show();
                return;
            }

            attemptLogin(email, password);
        });
    }

    private void attemptLogin(String username, String enteredPassword) {
        executor.execute(() -> {
            try {
                UserApi api = new UserApi();
                User user = api.readUser(username);

                runOnUiThread(() -> {
                    if (user == null) {
                        Toast.makeText(this, "User not found", Toast.LENGTH_SHORT).show();
                        return;
                    }
                    System.out.println("DEBUG usertype = " + user.usertype);

                    if (!"customer".equalsIgnoreCase(user.usertype)) {
                        Toast.makeText(this, "That account is not a customer account", Toast.LENGTH_SHORT).show();
                        return;
                    }


                    if (!TextUtils.isEmpty(user.password) && !enteredPassword.equals(user.password)) {
                        Toast.makeText(this, "Incorrect password", Toast.LENGTH_SHORT).show();
                        return;
                    }

                    Toast.makeText(this, "Logged in", Toast.LENGTH_SHORT).show();

                    Intent i = new Intent(this, CustomerDashboardActivity.class);
                    i.putExtra("USERNAME", user.username);
                    i.putExtra("DISPLAY_NAME", user.firstname);
                    startActivity(i);
                    finish();
                });

            } catch (Exception e) {
                runOnUiThread(() ->
                        Toast.makeText(this, "API error: " + e.getMessage(), Toast.LENGTH_LONG).show()
                );
            }
        });
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        executor.shutdown();
    }
}

File: app/src/main/java/com/example/restaurantapp/CustomerLoginActivity.kt
package com.example.restaurantapp

import android.content.Intent
import android.graphics.Color
import android.os.Bundle
import android.text.SpannableString
import android.text.Spanned
import android.text.method.LinkMovementMethod
import android.text.style.ClickableSpan
import android.text.style.ForegroundColorSpan
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.core.graphics.toColorInt

class CustomerLoginActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_customer_login)

        val backButton = findViewById<ImageView>(R.id.btnBack)
        val emailInput = findViewById<EditText>(R.id.edtCustomerEmail)
        val passwordInput = findViewById<EditText>(R.id.edtCustomerPassword)
        val continueButton = findViewById<Button>(R.id.btnCustomerContinue)
        val createAccountText = findViewById<TextView>(R.id.txtCreateAccount)

        // ✅ Make only "here" clickable & blue
        val fullText = "Not a user? Create account here"
        val spannable = SpannableString(fullText)

        val clickablePart = object : ClickableSpan() {
            override fun onClick(widget: View) {
                startActivity(Intent(this@CustomerLoginActivity, CreateAccountActivity::class.java))
            }
        }

        val startIndex = fullText.indexOf("here")
        val endIndex = startIndex + "here".length

        spannable.setSpan(clickablePart, startIndex, endIndex, Spanned.SPAN_EXCLUSIVE_EXCLUSIVE)
        spannable.setSpan(
            ForegroundColorSpan("#007AFF".toColorInt()),
            startIndex,
            endIndex,
            Spanned.SPAN_EXCLUSIVE_EXCLUSIVE
        )

        createAccountText.text = spannable
        createAccountText.movementMethod = LinkMovementMethod.getInstance()
        createAccountText.highlightColor = Color.TRANSPARENT // No background on tap

        // ✅ Login action
        continueButton.setOnClickListener {
            val email = emailInput.text.toString()
            val password = passwordInput.text.toString()

            if (email.isNotEmpty() && password.isNotEmpty()) {
                Toast.makeText(this, "Login successful", Toast.LENGTH_SHORT).show()
                startActivity(Intent(this, CustomerDashboardActivity::class.java))
            } else {
                Toast.makeText(this, "Please enter email and password", Toast.LENGTH_SHORT).show()
            }
        }

        // ✅ Back button
        backButton.setOnClickListener {
            startActivity(Intent(this, LoginActivity::class.java))
        }
    }
}
